### ISSUE TITLE
Expose MySqlDbType

### ISSUE THREAD

**Issue by bgrainger** [mysql-net/MySqlConnector#362]
The integer values of `MySqlDbType` are exposed by Connector/NET in `GetSchemaTable` (#307) and `GetSchema` (#361). `MySqlParameter` also has a `MySqlDbType` property.

Clients may be relying on the presence of this type in the public API and on its exact values.

**Comment by Thorium** [comment]
Would it be enough to rename `MySql.Data.Serialization.ColumnType` to `MySql.Data.MySqlClient.MySqlDbType` and modify the visiblity from `internal` to `public` with Visual Studio rename tool?

Or do the internal types have to match the MySql.Data types:

```csharp
	public enum MySqlDbType
	{
		Decimal,
		Byte,
		Int16,
		Int24 = 9,
		Int32 = 3,
		Int64 = 8,
		Float = 4,
		Double,
		Timestamp = 7,
		Date = 10,
		Time,
		DateTime,
		[Obsolete("The Datetime enum value is obsolete.  Please use DateTime.")]
		Datetime = 12,
		Year,
		Newdate,
		VarString,
		Bit,
		JSON = 245,
		NewDecimal,
		Enum,
		Set,
		TinyBlob,
		MediumBlob,
		LongBlob,
		Blob,
		VarChar,
		String,
		Geometry,
		UByte = 501,
		UInt16,
		UInt24 = 509,
		UInt32 = 503,
		UInt64 = 508,
		Binary = 600,
		VarBinary,
		TinyText = 749,
		MediumText,
		LongText,
		Text,
		Guid = 800
	}
```

**Comment by bgrainger** [comment]
`ColumnType` (an enum expressing values used in the MySQL protocol) should be kept separate from `MySqlDbType` (a separate enum used in the public API, that must maintain compatibility with Connector/NET).

A lot of the values will probably overlap (because MySql.Data may conflate the two enums).

**Comment by Thorium** [comment]
Yes I noticed MySqlDbType is missing column types like int and char. They are valid types however, that's why they need to be in GetSchema()

**Comment by bgrainger** [comment]
Implemented in [0.29.0](https://github.com/mysql-net/MySqlConnector/releases/tag/0.29.0).
